### ISSUE TITLE
apidoc buggy on linux, not windows

### ISSUE THREAD

**Issue by nikokio** [sphinx-doc/sphinx#3952]
Subject: When generating APIDOC, some stange "\" appears since release 1.5.4

### Problem
- Title of modules have a "\" in front of dots and underscore in title of modules

#### Procedure to reproduce the problem
```
sphinx-apidoc --separate --maxdepth 2 --doc-project "my project" --output-dir "/path"
```

#### Error logs / results
```
namespace1\.namespace2\.namespace3 module 
========================================= 
 
.. automodule:: namespace1.namespace2.namespace3
    :members: 
    :undoc-members: 
    :show-inheritance: 
```

#### Expected results
Good result should be :
```
namespace1.namespace2.namespace3 module 
======================================= 
 
.. automodule:: namespace1.namespace2.namespace3
    :members: 
    :undoc-members: 
    :show-inheritance: 
### Reproducible project / your project
```

### Environment info
- OS: Linux
- Python version: 3.5.2
- Sphinx version: 1.6.2

I notice it works well with 1.5.3 but from 1.5.4 up to 1.6.3, it doesn't
I also notice it works well on Windows...

**Comment by tk0miya** [comment]
Okay, I understand.
This came from #3581 . The back-slashes are used for escaping. It's not bug.

I understand it is not easy to read. so the escaping function should be improved.

**Comment by tk0miya** [comment]
Fixed by #4511.
Thank you for reporting
